### PR TITLE
Add String.Create span-based method

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/corefx/issues/22380

### CODE REVIEW

File: src/mscorlib/src/System/String.cs [hunk 1]
```diff
@@ -684,6 +685,24 @@ private unsafe string CtorReadOnlySpanOfChar(ReadOnlySpan<char> value)
             return result;
         }
 
+        public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)
```
Comment (justinvp) [inline src/mscorlib/src/System/String.cs]: Throw `ArgumentNullException` if `action` is null?

Comment (stephentoub) [reply src/mscorlib/src/System/String.cs]: Yup.

**Comment by stephentoub** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please
@dotnet-bot test OSX 10.12 x64 Checked Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please (hanging doing "Cloning the remote Git repository")

**Pull request merged**